samples/sample_common/src/vaapi_allocator.cpp

```diff
@@ -141,6 +141,102 @@ mfxStatus vaapiFrameAllocator::Close()
     return BaseFrameAllocator::Close();
 }
 
+mfxStatus vaapiFrameAllocator::ReallocImpl(mfxMemId mid, const mfxFrameInfo *info, mfxU16 memType, mfxMemId *midOut)
```

Comment:
it's almost a copy/past of AllocImpl(). I mean code duplication. It should be avoided. 
I'd suggest to define a common part of AllocImpl and ReallocImpl and move it to a separate private function.

Reply:
I can move the part of get VA FOURCC into a common private function. Thanks.